src/actions/proxy.ts

```diff
@@ -0,0 +1,161 @@
+      }
+    }
+
+    proxyCount = async (): Promise<BN> => {
+      const res: { 0: BN }  = await executeAndRetry(this.contracts['PROXY_REGISTRY'].count, []);
+      return res[0];
+    }
+
+    checkProxyExistence = async (borrowerAddr: string) => {
+      let proxy;
+      const count = (await this.proxyCount()).toNumber();
+      for (let i = 0; i < count; i += 1) {
+        const ownerBN = await this.getAccessTokenOwner(i.toString())
+        if (ownerBN && ethers.utils.getAddress(ownerBN.toString()) === ethers.utils.getAddress(borrowerAddr)) {
+          const proxyAddress = await this.getProxy(i.toString());
+          proxy = proxyAddress;
+        }
+      }
+      return proxy;
+    }
+
+    proxyCreateNew = async (borrowerAddr: string) => {
+      const accessToken = await this.buildProxy(borrowerAddr);
+      return await this.getProxy(accessToken);
+    }
+
+    proxyTransferIssue = async (proxyAddr: string, tokenId: string)  => {
+      const proxy: any = this.eth.contract(this.contractAbis['PROXY']).at(proxyAddr);
+
+      const encoded = abiCoder.encodeFunctionCall({
+        name: 'transferIssue',
+        type :'function',
+        inputs: [
+          { type: 'address', name: 'shelf' },
+          { type: 'address', name: 'registry' },
+          { type: 'uint256', name: 'token' }]},   [this.contracts['SHELF'].address, this.contracts['COLLATERAL_NFT'].address, tokenId],
+      );
+
+      const txHash = await executeAndRetry(proxy.execute, [this.contracts['ACTIONS'].address, encoded, this.ethConfig]);
+      console.log(`[Proxy Transfer Issue Loan] txHash: ${txHash}`);
+      return await waitAndReturnEvents(this.eth, txHash, this.contractAbis['PROXY'], this.transactionTimeout);
+    }
+
+    proxyLockBorrowWithdraw = async (proxyAddr: string, loanId: string, amount: string, usr: string) => {
+      const proxy: any = this.eth.contract(this.contractAbis['PROXY']).at(proxyAddr);
+      const encoded = abiCoder.encodeFunctionCall({
+        name: 'lockBorrowWithdraw',
+        type :'function',
+        inputs: [
+                                                      { type: 'address', name: 'shelf' },
+                                                      { type: 'uint256', name: 'loan' },
+                                                      { type: 'uint256', name: 'amount' },
+                                                      { type: 'address', name: 'usr' }]},
+                                                  [this.contracts['SHELF'].address, loanId, amount, usr],
+      );
+      const txHash = await executeAndRetry(proxy.execute, [this.contracts['ACTIONS'].address, encoded, this.ethConfig]);
+      console.log(`[Proxy Lock Borrow Withdraw] txHash: ${txHash}`);
+      return await waitAndReturnEvents(this.eth, txHash, this.contractAbis['PROXY'], this.transactionTimeout);
+    }
+
+    proxyRepayUnlockClose = async (proxyAddr: string, tokenId: string, loanId: string, amount: string) => {
```

Comment:
still old interface is used. we are not passing the amount to this function anymore. 
--> repayUnlockClose(ShelfLike shelf, PileLike pile, NFTLike registry, uint token, ERC20Like erc20, uint loan)